code/obj/item/grab.dm

```diff
@@ -743,6 +746,15 @@
 			playsound(assailant.loc, 'sound/impact_sounds/Generic_Shove_1.ogg', 50, 1, 0, 1.5)
 		qdel(src)
 
+	setProperty(propId, propVal)
+		var/datum/objectProperty/equipment/P = ..()
+		if(istype(P))
+			P.updateMob(src, src.assailant, propVal)
+
+	delProperty(propId)
+		var/datum/objectProperty/equipment/P = ..()
+		if(istype(P))
+			P.removeFromMob(src, src.assailant)
```

Comment:
`removeFromMob` actually takes the current property value and I can imagine situations where it might be relevant so it should probably get passed as an argument. Though I admit that this doesn't look elegant at all.
```suggestion
		var/value = getProperty(propId)
		var/datum/objectProperty/equipment/P = ..()
		if(istype(P))
			P.removeFromMob(src, src.assailant, value)
```

Reply:
unless i'm being dumb can just get the value of P directly instead of needing to call getProperty()?

Reply:
How?

Reply:
You can't do `var/value = src.properties[P]` because at that point the property is already deleted.
`removeFromMob` can't get it inside itself because of the same reason.

Reply:
As a matter of fact: I was being dumb!